cpp/modmesh/mesh/plot3d2mesh.cpp

```diff
@@ -0,0 +1,107 @@
+#include "StaticMesh.hpp" // Include modmesh headers
+#include <fstream>
+
+namespace modmesh
+{
+
+// The parser logic goes here
+void ParsePlot3DFile(const std::string & filename)
+{
+    // Open file, handle errors
+    std::ifstream p3d_File(filename);
+    if (!p3d_File.is_open())
+    {
+        throw std::runtime_error("Cannot open file: " + filename);
+    }
+
+    // Parse the file and populate the StaticMesh
+    // This will involve reading the coordinates and other data
+    // from the file and using StaticMesh methods and properties
+    // to store this data.
+
+    uint8_t ndim = 3;
+    modmesh::uint_type nnode = 0;
+    modmesh::uint_type nface = 0;
+    modmesh::uint_type ncell = 0;
+
+    auto mesh = modmesh::StaticMesh::construct(ndim, nnode, nface, ncell);
+    (*mesh).set_ndim(static_cast<int>(3));
+    std::cout << "Dim is " << static_cast<int>((*mesh).ndim()) << std::endl;
+
+    p3d_File >> ncell;
+    (*mesh).set_ncell(ncell);
+    std::cout << "ncell is: " << (*mesh).ncell() << std::endl;
+
+    int Imax, Jmax, Kmax;
+
+    for (size_t i = 0; i < (*mesh).ncell(); i++)
+    {
+        p3d_File >> Imax >> Jmax >> Kmax;
+        std::cout << "Imax: " << Imax << ", Jmax: " << Jmax << ", Kmax: " << Kmax << std::endl;
+        nnode += Imax * Jmax * Kmax;
+    }
+
+    (*mesh).set_nnode(nnode);
+    std::cout << "nnode is: " << (*mesh).nnode() << std::endl;
+    mesh->ndcrd().remake(modmesh::small_vector<size_t>{(*mesh).nnode(), 3}, 0);
+
+    // Reading coordinates and populating 'ndcrd'
+    for (size_t i = 0; i < (*mesh).nnode(); ++i)
+    {
+        double x;
+        p3d_File >> x;
+        mesh->ndcrd(i, 0) = x;
+    }
+    for (size_t i = 0; i < (*mesh).nnode(); ++i)
+    {
+        double y;
+        p3d_File >> y;
+        mesh->ndcrd(i, 1) = y;
+    }
+    for (size_t i = 0; i < (*mesh).nnode(); ++i)
+    {
+        double z;
+        p3d_File >> z;
+        mesh->ndcrd(i, 2) = z;
+        std::cout << "ndcrd(" << i << ",2): " << mesh->ndcrd(i, 2) << std::endl;
+    }
+
+    p3d_File.close();
+}
+
+} /* end namespace modmesh */
+
+int main(int argc, char * argv[])
```

Comment:
Is this `main()` function for testing?

Reply:
Right! This main() function is just for my local testing when developing. I'll remove it when all things are done.